app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/network/MarvelRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.network
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.Provider
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.RetrofitApi
+import io.reactivex.Single
+
+interface MarvelRepository {
+
+    fun getCharacters(searchQuery: String?): Single<List<MarvelCharacter>>
+
+    class Impl(val api: MarvelApi) : MarvelRepository {
```

Comment:
this is weird, interface with single implementation is rather anti-pattern, I would just create class MarvelRepository